src/Symfony/Component/Console/Command/InterruptibleTrait.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Exception\RuntimeException;
+
+/**
+ * This trait adds ability to interrupt safely long-running or endless commands in case "^C" is pressed or "$ kill -2".
+ *
+ * @author Maksym Slesarenko <maks.slesarenko@gmail.com>
+ *
+ * @example
+ * protected function execute(InputInterface $input, OutputInterface $output)
+ * {
+ *     while (true) {
+ *         $this->isInterrupted(true);
```

Comment:
This usage seems wrong to me. I'd prefer to see two methods. I'd suggest to rename your `isInterrupted(true)` use case to something like a `throwIfInterupted()` call.